Apps/Database/src/Delegates/DBWalletDelegate.cs

```diff
@@ -0,0 +1,212 @@
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.Database.Delegates
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Linq;
+    using System.Text.Json;
+    using HealthGateway.Database.Constants;
+    using HealthGateway.Database.Context;
+    using HealthGateway.Database.Models;
+    using HealthGateway.Database.Wrapper;
+    using Microsoft.EntityFrameworkCore;
+    using Microsoft.Extensions.Logging;
+
+    /// <inheritdoc />
+    [ExcludeFromCodeCoverage]
+    public class DBWalletDelegate : IWalletDelegate
+    {
+        private readonly ILogger logger;
+        private readonly GatewayDbContext dbContext;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DBWalletDelegate"/> class.
+        /// </summary>
+        /// <param name="logger">Injected Logger Provider.</param>
+        /// <param name="dbContext">The context to be used when accessing the database.</param>
+        public DBWalletDelegate(
+            ILogger<DBFeedbackDelegate> logger,
+            GatewayDbContext dbContext)
+        {
+            this.logger = logger;
+            this.dbContext = dbContext;
+        }
+
+        /// <inheritdoc />
+        public DBResult<WalletConnection> GetConnection(Guid id)
+        {
+            DBResult<WalletConnection> result = new ()
+            {
+                Status = DBStatusCode.NotFound,
+            };
+            WalletConnection? connection = this.dbContext.WalletConnection
+                                    .Where(p => p.Id == id)
+                                    .FirstOrDefault();
+            if (connection != null)
+            {
+                result.Status = DBStatusCode.Read;
+                result.Payload = connection;
+            }
+
+            return result;
+        }
+
+        /// <inheritdoc />
+        public DBResult<WalletConnection> GetConnection(string userProfileId)
+        {
+            DBResult<WalletConnection> result = new ()
+            {
+                Status = DBStatusCode.NotFound,
+            };
+            WalletConnection? connection = this.dbContext.WalletConnection
+                                    .Where(p => p.UserProfileId == userProfileId &&
+                                                p.Status != WalletConnectionStatus.Disconnected)
```

Comment:
I think that If disconnected connections are not going to be returned then we should add that to the docs of the method or reflect that in the method name itself. Also, it would be less confusing for the caller if both methods returned disconnected connections or not. Otherwise, the one that takes an Id returns values that the other one might not

Reply:
My thought around this was that the front-end would only ever want connected that were not in the disconnected state as that is the trigger to start the process over.  The backend portion of the code for status updates will always use the direct ID lookup.  This also allows us to always return a single result, otherwise the get by ProfileID has to return a list while the get by primary key will return a single.

Reply:
I think the problem is that that would break the abstraction by the delegate 'knowing' which services and how they are called.

Reply:
What about renaming this method to something like `GetActiveConnection`?  Then the front-end would still be able to have a way to grab the one connection it cares about and the delegate would have a reason to be filtering on the status.

Reply:
I created 10650 to update this.